src/main.rs

```diff
@@ -35,7 +36,8 @@ async fn main() -> Result<(), Box<dyn Error>> {
         .merge(routes::user_routes::routes(State(app_state.clone())))
         .merge(routes::password_routes::routes(State(app_state.clone())))
         .merge(routes::session_routes::routes(State(app_state.clone())))
-        .layer(middleware::map_response(main_response_mapper));
+        .layer(middleware::map_response(main_response_mapper))
+        .route_layer(middleware::from_fn(with_api_key));
```

Comment:
What's the difference between`layer` and `router_layer`

Reply:
In Axum, `layer` applies middleware to the entire router, affecting all routes within that router, while `route_layer` applies middleware to a specific route, affecting only that route. Use `layer` for global middleware and `route_layer` for route-specific middleware.

But we can use layer also here for consistancy, so i changed it to layer only.